### PR TITLE
[in_app_purchase] Configured example app to use StoreKit Testing on iOS 14

### PR DESCRIPTION
**Pull request by mvanbeusekom**
Configured example app to use StoreKit Testing on iOS 14 and higher. 

- Fixes flutter/flutter#79286

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme [hunk 1]
```diff
@@ -69,6 +69,9 @@
             ReferencedContainer = "container:Runner.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
+      <StoreKitConfigurationFileReference
+         identifier = "../Runner/Configuration.storekit">
+      </StoreKitConfigurationFileReference>
```
[the remark this answers is not in the document]
Comment (jmagman) [reply packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:74]: I see the same think from Xcode and `flutter run --debug`:
Screen Shot 2021-03-30 at 12 36 00 PM

What are the repro steps to see it not working?

[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:74]: I started the example app from Android Studio (run not debug). I guess the `--debug` makes the difference.

[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:74]: Hmm I will look into it again. I was also running on a simulator.

I did notice that if I run it first with Xcode and afterwards from Android Studio it worked fine. However if I start a fresh simulator (which didn't have the app installed) and run it with Android Studio I run into some exceptions. 

It is kinda late over here so I will do some more testing tomorrow and if I run into issues I will make a detailed report.

[the remark this answers is not in the document]
Comment (jmagman) [reply packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:74]: Hm it's weird, if I run it once from Xcode it works, and then subsequently from the command line it also works, so Xcode is laying something else down.  The bundles look identical.  The entitlements are the same.

[the remark this answers is not in the document]
Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:74]: Yes indeed, Xcode is doing some magic regarding the `Configuration.storekit` file in my opinion. I followed the [Setting Up StoreKit Testing in Xcode](https://developer.apple.com/documentation/xcode/setting_up_storekit_testing_in_xcode?language=objc) article which describe to enable StoreKit Testing in Xcode through the following steps:

> 1. Click the scheme to open the scheme menu; choose Edit Scheme.
> 2. In the scheme editor, choose the Run action.
> 3. Click Options in the action settings.
> 4. For the StoreKit Configuration option, select a configuration file.

I have the feeling that Xcode somehow parses the `Configuration.storekit` and registers the products configured in the file separate from the App on the simulator (some form of offline StoreKit cache). If you make changes to the `Configuration.storekit` file and run the App using `flutter run` the changes are not reflected in the App it self. If you run the App using Xcode the changes are correctly reflected in the App.

**Review by cyanglaz — APPROVED** [review]
LGTM! The storekit testing doesn't seem to be working in command line, which will delay our plan to make XCUITests for IAP. However, this PR is a big improvement for the example app and there's no regression, so we should land it!

**Comment by kpradeepkumarreddy** [comment]
StoreKit is for testing with the data that is configured locally. How to test with the consumables created  on app store connect using simulator ?

**Comment by mvanbeusekom** [comment]
> StoreKit is for testing with the data that is configured locally. How to test with the consumables created  on app store connect using simulator ?

Unfortunately you can't. If you want to test your consumables configured in the AppStore you will have to test on a real device.

**Comment by cyanglaz** [comment]
@mvanbeusekom Maybe we should add this information in our docs too?

**Comment by sneglia9191** [comment]
Is there any additional information on this?  It does appear StoreKit testing will work using Android Studio, only after executed from Xcode one time.  Is this the assumed workaround, or will a fix be implemented to initially work from Android Studio?

**Comment by jmagman** [comment]
>  It does appear StoreKit testing will work using Android Studio, only after executed from Xcode one time. Is this the assumed workaround, or will a fix be implemented to initially work from Android Studio?

@sneglia9191 can you file a new issue describing what you're seeing? I'm unfamiliar with that problem.

**Pull request merged**